src/app/views/shared/dynamic-library-examples/example-components/highcharts/highcharts.component.ts

```diff
@@ -50,170 +51,218 @@ export class HighchartsComponent implements OnInit {
     };
   }
 
-  onClickFilterDataset(value: string) {
-    this.dataIsLoading = true;
-    this.dataIsLoaded = false;
-
-    switch (value) {
-      case '2017-2021':
-        this.getDodsfallEtterAarsak_2017_2021_Dummyfilter();
-        break;
-      case '2017':
-        this.getDodsfallEtterAarsak_2017_Dummyfilter();
-        break;
-      case 'hjerteinfarkt-mann':
-        this.getDodsfallEtterAarsak_2017_2021_Hjerteinfarkt_Mann_Dummyfilter();
-        break;
-    }
+  onSelectMockData(value: string) {
+    this.selectMockData(value);
   }
 
   onMetadataButtonClick() {
     this.viewportScroller.scrollToAnchor('om-dataene');
   }
 
-  private getDodsfallEtterAarsak_2008_2018() {
-    this.highchartsDataService.getData(MockData.DodsfallEtterAarsak_2008_2018).subscribe({
+  private getData(mockDataId: number, diagramOptions: FhiDiagramOptions) {
+    this.dataIsLoading = true;
+    this.dataIsLoaded = false;
```

Comment:
on the line above you set `dataIsLoading` to `true` and here you change it without any logic in between. Is it supposed to be like this?

Reply:
Two properties, maybe poorly named, but do not change them in this PR.